src/ReflectionClassConstant.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Parser Reflection API
+ *
+ * @copyright Copyright 2015, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+declare(strict_types=1);
+
+namespace Go\ParserReflection;
+
+use Go\ParserReflection\ValueResolver\NodeExpressionResolver;
+use PhpParser\Node\Const_;
+use PhpParser\Node\Stmt\ClassConst;
+use PhpParser\Node\Stmt\ClassLike;
+use \ReflectionClassConstant as BaseReflectionClassConstant;
+
+class ReflectionClassConstant extends BaseReflectionClassConstant
+{
+    /**
+     * Concrete class constant node
+     *
+     * @var ClassConst
+     */
+    private $classConstantNode;
+
+    /**
+     * @var Const_
+     */
+    private $constNode;
+
+    /**
+     * Name of the class
+     *
+     * @var string
+     */
+    private $className;
+
+    /**
+     * Parses class constants from the concrete class node
+     *
+     * @param ClassLike $classLikeNode Class-like node
+     * @param string $reflectionClassName FQN of the class
+     *
+     * @return array|ReflectionClassConstant[]
+     */
+    public static function collectFromClassNode(ClassLike $classLikeNode, $reflectionClassName)
+    {
+        $classConstants = [];
+
+        foreach ($classLikeNode->stmts as $classLevelNode) {
+            if ($classLevelNode instanceof ClassConst) {
+                foreach ($classLevelNode->consts as $const) {
+                    $classConstName = $const->name->toString();
+                    $classConstants[$classConstName] = new ReflectionClassConstant(
+                        $reflectionClassName,
+                        $classConstName,
+                        $classLevelNode,
+                        $const
+                    );
+                }
+            }
+        }
+
+        return $classConstants;
+    }
+
+    /**
+     * Initializes a reflection for the class constant
+     *
+     * @param string $className Name of the class
+     * @param string $classConstantName Name of the class constant to reflect
+     * @param ClassConst $classConstNode ClassConstant definition node
+     * @param Const_|null $constNode Concrete const definition node
+     */
+    public function __construct(
+        $className,
+        $classConstantName,
```

Comment:
Missing argument type `string` for first two arguments

Reply:
Done!